```{r,message=FALSE,echo=FALSE}
  library("knitcitations")
  bib <- read.bibtex("bibtexlib.bib")
  library("knitr")
  cite_options(citation_format = "pandoc", max.names = 3, style = "html", hyperlink = "to.doc")
  opts_chunk$set(tidy = FALSE, message = FALSE, warning = FALSE)
```
---
title: "Getting ready to use R"
subtitle: "*ZSL Foster and NJ Gr&uuml;nwald*"
---

R provides a unique environment for performing population genetic analyses. You
will particularly enjoy not having to switch data formats and operating systems
to execute a series of analyses, as was the case until now. Furthermore, R
provides graphing capabilities that are ready for use in publications, with only
a little bit of extra effort. But first, let's install R, install an integrated
development environment, open R, and load R packages.

Installing R
-----

1. Download and install the [R](http://www.r-project.org) statistical computing
   and graphing environment. This works cross-platform on Windows, OS X and
   Linux operating systems.

2. Download and install the free, open source edition of the [RStudio Desktop](http://www.rstudio.com/ide/) integrated development environment (IDE) that we recommend.

Installing the required packages
----

The following packages are utilized in this primer:

1. [vegan](https://cran.r-project.org/web/packages/vegan/index.html)
2. [metacoder](https://cran.r-project.org/web/packages/metacoder/index.html)
3. [phyloseq](https://joey711.github.io/phyloseq/install.html)
4. [taxa](https://cran.r-project.org/web/packages/taxa/index.html)
5. [ggplot2](http://cran.r-project.org/web/packages/ggplot2/index.html)

More R resources can be found here: https://microsud.github.io/Tools-Microbiome-Anlaysis/. 

Use the following script to install these packages:

```{r, eval = FALSE}
install.packages(c("vegan", "metacoder", "ggplot2", "taxa"), repos = "http://cran.rstudio.com", dependencies = TRUE)
```

We wrote and actively maintain *metacoder* `r citep(bib["foster2017metacoder"])` and it
is heavily relied upon in this primer. You can think of an R package as a library of 
functions written and curated by someone in the R user
community, which you can be loaded into R for use.

Congratulations. You should now be all set for using R. Loading data and
conducting your first analysis will be the topic of the next chapter. But before
we go there lets provide a few useful resources.

A quick introduction to R using RStudio
----

Next, let's review some of the basic features and functions of R. To start R,
open the RStudio application from your programs folder or start menu. This will
initialize your R session. To exit R, simply close the RStudio application.

> Note that R is a case sensitive language!

Let's get comfortable with R by submitting the following command on the command
line (where R prompts you with a `>` in the lower left RStudio window pane) that
will retrieve the current working directory on your machine:

```{r, eval=FALSE}
getwd() # this command will print the current working directory
```

> Note that the symbol '#' is used to add comments to your code and you just
> type `getwd()` after the ">".

Our primer is heavily based on the *poppr* and *adegenet* packages. To get help
on any of their functions type a question mark before the empty function call as
in:

```{r, eval=FALSE}
?mlg # open the R documentation of the function mlg()
```

To quit R you can either use the <kbd>RStudio > Quit</kbd> pull-down menu
command or execute <kbd>âŒ˜ + Q</kbd> (OS X) or <kbd>ctrl + Q</kbd> (PC).


Packages and getting help
----

One way that R shines above other languages for analysis is the fact that
R packages in CRAN are all documented. Help files are written in HTML and give
the user a brief overview of:

 - the purpose of a function,
 - the parameters it takes,
 - the output it yields,
 - and some examples demonstrating its usage.

To see all of the help topics in a package, you can simply type:

```{r, eval = FALSE}
help(package = "metacoder") # Get help for a package.
help(extract_taxonomy)      # Get help for the extract_taxonomy function.
?extract_taxonomy           # same as above.
??unite                # Search for functions that have the keyword multilocus.
```

Some packages include vignettes that can have different formats such as being
introductions, tutorials, or reference cards in PDF format. You can look at a
list of vignettes in all packages by typing:

```{r, eval=FALSE}
browseVignettes()                     # see vignettes from all packages
browseVignettes(package = 'metacoder')    # see vignettes from a specific package.
```

Next, consider browsing  Appendix 3 on "Introduction to R" if you are not yet
familiar with R and RStudio. Otherwise, you are now ready to think about
formatting and loading population genetic data into R.

References
----------

<!--- ref --->
