---
title: "Data quality control"
output: html_document
---

```{r setup, include=FALSE}
source("style.R")
```



Sequencing technologies all have some amount of error.
Some of this error happens during PCR and some happens during sequencing.
There are lots of ways to filter out errors including: 

* Removing sequences/bases that the sequencer indicates are low quality.
This information is typically in the .fastq files returned by the sequencer.
There are many programs, such as "trimmomatic", that can use the quality scores in these files to filter out low-quality sequences.
* Clustering similar sequences together. Many erroneous sequences are very similar to the true sequences, so they will be incorporated into the same OTU as the true sequence. In that sense, clustering into OTUs will hide some sequencing error.
* Remove of chimeric sequences. "Chimeras" are more significant errors that occur during PCR when an incomplete amplicon acts as a primer for a different template in a subsequent cycle. Most amplicon metagenomic pipelines (e.g. QIIME, mothur, usearch) have functions to detect and remove chimeras.
* Removing low-abundance sequences. Most of the time erroneous sequences will occur much less often than the true sequence. Simply removing any unique sequence / OTU that appears less than some minimum number of times is an effective way to remove these errors. OTUs represented by only a single sequence are commonly called "singletons". So when you hear things like "singletons were removed" and means all OTUs with only a single sequence were removed.
* Rarefying read counts. Usually, we try to make each sample in a sequencing run have a the same number of reads sequenced, but that does not always happen. When a sample has many more reads than another sample, its apparent diversity can be artificially inflated since rare taxa are more likely to be found. To avoid this, the reads are subsampled to a fixed number or "rarefied". 
* Converting counts to presence/absence. Due to PCR biases and the nature of compositional data, many researchers advocate not useing read count information at all. Instead, they reccomend converting counts to simply "present" or "absent". 

Since we are only working with abundance matrices in these tutorials, the first three types of quality control have already been done (ideally), but we can still remove low-abundance OTUs and rarefy the counts of the remaining OTUs.

## Load example data

If you are starting the workshop at this section, or had problems running code in a previous section, use the following to load the data used in this section.
If `obj` and `sample_data` are already in your environment, you can ignore this and proceed.

```{r}
load("filtered_data.Rdata")
```

## Removing low-abundance counts

The easiest way to get rid of some error in your data is to throw out any count information below some threshold.
The threshold is up to you; removing singletons or doubletons is common, but lets be more conservative and remove any counts less than 10.

```{r}
library(metacoder)
obj$data$otu_counts <- zero_low_counts(obj, "otu_counts", min_count = 10)
```

That set all read counts less than 10 to zero.
It did not filter out any OTUs or their assocaited taxa however.
We can do that using `filter_obs`, which is used to filter data assoicated with a taxonomy in a `taxmap` object.
First lets find which OTUs now have now reads assocaited with them.

```{r}
no_reads <- rowSums(obj$data$otu_counts[, sample_data$SampleID]) == 0
sum(no_reads) # when `sum` is used on a TRUE/FALSE vector it counts TRUEs
```

So now `r sum(no_reads)` of `r nrow(obj$data$otu_counts)` have no reads.
We can remove them and the taxa they are assocaited with like so: 

```{r}
obj <- filter_obs(obj, "otu_counts", ! no_reads, drop_taxa = TRUE)
print(obj)
```


## Rarefaction

Rarefaction is used to simulate even sampling depth.
Even sampling depth is important for at least two reasons:

* When comparing diversity of samples, a higher sampling depth can inflate appparent diverstiy since more rare species are likely to be found. The effect of this will depend on the diversity index used.
* When comparing the similarity of samples, the presence of rare species due to higher sampling depth in one sample but not another can make the two samples appear more differnet than they actually are.

Therefore, when comparing the diversity or similarity of samples, it is important to rarefy, or subsample, to a constant depth.
Typically, the depth chosen is the minimum sample depth.
If the minimum depth is very small, the samples with the smallest depth can be removed and the minimum depth of the remaining samples can be used.

Lets take a look at the distribution of read depths of our samples:

```{r}
hist(colSums(obj$data$otu_counts[, -1]))
```

We have a minimum depth of `r min(colSums(obj$data$otu_counts[, -1]))`, a median of `r as.integer(median(colSums(obj$data$otu_counts[, -1])))` and a maximum depth of `r max(colSums(obj$data$otu_counts[, -1]))`.
We could try to remove one or two of the samples with the smallest depth, since it seems like a waste to throw out so much data, but for this tutorial we will just rarefy to the minimum of `r min(colSums(obj$data$otu_counts[, -1]))` reads.
The `vegan` package implements many functions to help with rarefying data, but we will use a function from `metacoder` that calls the functions from `vegan` and reformats the input and ouputs to make them easier to use with the way our data is formatted.

```{r}
obj$data$otu_rarefied <- rarefy_obs(obj, "otu_counts")
print(obj)
```


## Rarefaction curves


```{r}
library(vegan)
rarecurve(t(obj$data$otu_counts[1:10, -1]), step = 20,
          sample = min(colSums(obj$data$otu_counts[, -1])), col = "blue", cex = 0.6)
```

## Converting to presence/absence

## Saving the data for the next section

In the next section we will work on analyzing this data set.
Don't worry about running the following code; it is just there to let us use the same data set on multiple website pages.

```{r}
save(obj, sample_data, file = "clean_data.Rdata")
```


```{r, child="_sessioninfo.Rmd"}
```