<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-52691601-4"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-52691601-4');
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analysis of Microbiome Data in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="02--quick_example.html">Example analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00--required_software.html">Required software</a>
    </li>
    <li>
      <a href="00--required_data.html">Required datasets</a>
    </li>
    <li>
      <a href="03--parsing.html">Getting data into R</a>
    </li>
    <li>
      <a href="04--manipulating.html">Manipulating taxonomic data</a>
    </li>
    <li>
      <a href="05--plotting.html">Plotting taxonomic data</a>
    </li>
    <li>
      <a href="06--quality_control.html">Data quality control</a>
    </li>
    <li>
      <a href="07--diversity_stats.html">Diversity and Ordination</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00--intro_to_r.html">Introduction to R</a>
    </li>
    <li>
      <a href="00--intro_to_rstudio.html">Introduction to RStudio</a>
    </li>
    <li>
      <a href="00--glossary.html">Glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r">Website source code</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="diversity-statistics" class="section level1">
<h1>Diversity statistics</h1>
<div id="load-example-data" class="section level2">
<h2>Load example data</h2>
<p>If you are starting the workshop at this section, or had problems running code in a previous section, use the following to load the data used in this section. You can download the “clean_data.Rdata” file <a href="clean_data.Rdata" download="clean_data.Rdata">here</a>. If <code>obj</code> and <code>sample_data</code> are already in your environment, you can ignore this and proceed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;clean_data.Rdata&quot;</span>)</code></pre></div>
</div>
<div id="measures-of-diversity" class="section level2">
<h2>Measures of diversity</h2>
<p>Diversity in the ecological sense is intuitively understood as the complexity of a community of organisms. There are many ways to quantify this complexity so that we can compare communities objectively. The two main categories of methods are known as <strong>alpha diversity</strong> and <strong>beta diversity</strong> <span class="citation">(Whittaker 1960)</span>. Alpha diversity measures the diversity within a single sample and is generally based on the number and relative abundance of taxa at some rank (e.g. species or OTUs). Beta diversity also uses the number of relative abundance of taxa at some rank, but measures variation between samples. In other words, an alpha diversity statistic describes a single sample and a beta diversity statistic describes how two samples compare.</p>
<p>The <code>vegan</code> package is the main tool set used for calculating biological diversity statistics in R.</p>
</div>
<div id="alpha-within-sample-diversity" class="section level2">
<h2>Alpha (within sample) diversity</h2>
<p>Common alpha diversity statistics include:</p>
<ul>
<li><strong>Shannon</strong>: How difficult it is to predict the identity of a randomly chosen individual.</li>
<li><strong>Simpson</strong>: The probability that two randomly chosen individuals are the same species.</li>
<li><strong>Inverse Simpson</strong>: This is a bit confusing to think about. Assuming a theoretically community where all species were equally abundant, this would be the number of species needed to have the same Simpson index value for the community being analyzed.</li>
</ul>
<p>There are also some diversity indexes that take into account the taxonomic similarity of the species called “taxonomic diversity” and “taxonomic distinctness”, but we will not go into those.</p>
<p>The <code>diversity</code> function from the <code>vegan</code> package can be used to calculate the alpha diversity of a set of samples. Like other vegan functions, it assumes that samples are in rows, but they are in columns in our data, so we need to use the <code>MARGIN = 2</code> option. We also need to exclude the taxon ID column by subsetting the columns to only samples (i.e. all column besides the first one). Since alpha diversity is a per-sample attribute, we can just add this as a column to the sample data table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vegan)
sample_data<span class="op">$</span>alpha &lt;-<span class="st"> </span><span class="kw">diversity</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_rarefied[, sample_data<span class="op">$</span>SampleID],
                                 <span class="dt">MARGIN =</span> <span class="dv">2</span>,
                                 <span class="dt">index =</span> <span class="st">&quot;invsimpson&quot;</span>)
<span class="kw">hist</span>(sample_data<span class="op">$</span>alpha)</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-2-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Adding this as a column to the sample data table makes it easy to graph using <code>ggplot2</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
<span class="kw">ggplot</span>(sample_data, <span class="kw">aes</span>(<span class="dt">x =</span> Site, <span class="dt">y =</span> alpha)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_boxplot</span>()</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-3-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>We can use <a href ="00--glossary.html#analysis_of_variance_(anova)_anchor">analysis of variance (ANOVA)</a> to tell if at least one of the diversity means is different from the rest.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">anova_result &lt;-<span class="st"> </span><span class="kw">aov</span>(alpha <span class="op">~</span><span class="st"> </span>Site, sample_data)
<span class="kw">summary</span>(anova_result)</code></pre></div>
<pre class="r-output"><code>##              Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## Site          2  79081   39540   14.52 9.78e-07 ***
## Residuals   289 786895    2723                     
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
</code></pre>
<p>That tells us that there is a difference, but does not tell us which means are different. A <a href ="00--glossary.html#tukey's_honest_significant_difference_(hsd)_anchor">Tukey’s Honest Significant Difference (HSD)</a> test can do pairwise comparisons of the means to find this out. We will use the <code>HSD.test</code> function from the <code>agricolae</code> package since it provides grouping codes that are useful for graphing.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(agricolae)
tukey_result &lt;-<span class="st"> </span><span class="kw">HSD.test</span>(anova_result, <span class="st">&quot;Site&quot;</span>, <span class="dt">group =</span> <span class="ot">TRUE</span>)
<span class="kw">print</span>(tukey_result)</code></pre></div>
<pre class="r-output"><code>## $statistics
##    MSerror  Df     Mean       CV
##   2722.819 289 57.48091 90.77907
## 
## $parameters
##    test name.t ntr StudentizedRange alpha
##   Tukey   Site   3          3.33168  0.05
## 
## $means
##        alpha      std   r      Min      Max       Q25      Q50       Q75
## Jam 32.62941 31.37153  84 1.266110 101.3504  1.676679 32.45049  58.00689
## Mah 62.03473 49.53254 104 2.144132 153.5168 13.061531 58.63623 104.13730
## Sil 72.99945 66.28129 104 3.431795 224.0853 12.485001 37.78903 142.18317
## 
## $comparison
## NULL
## 
## $groups
##        alpha groups
## Sil 72.99945      a
## Mah 62.03473      a
## Jam 32.62941      b
## 
## attr(,"class")
## [1] "group"
</code></pre>
<p>Looking at the <code>tukey_result$groups</code> table it appears that the alpha diversity of sites “Sil” and “Mah” might not be different, but there is evidence that the diversity in site “Jam” is lower. We can add this information to the graph using the <code>tukey_result$groups$groups</code> codes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">group_data &lt;-<span class="st"> </span>tukey_result<span class="op">$</span>groups[<span class="kw">order</span>(<span class="kw">rownames</span>(tukey_result<span class="op">$</span>groups)),]
<span class="kw">ggplot</span>(sample_data, <span class="kw">aes</span>(<span class="dt">x =</span> Site, <span class="dt">y =</span> alpha)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(),
            <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">rownames</span>(group_data), <span class="dt">y =</span> <span class="kw">max</span>(sample_data<span class="op">$</span>alpha) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">label =</span> group_data<span class="op">$</span>groups),
            <span class="dt">col =</span> <span class="st">&#39;black&#39;</span>,
            <span class="dt">size =</span> <span class="dv">10</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Alpha diversity&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Site&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Alpha diversity index&quot;</span>)</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>So that takes care of comparing the alpha diversity of sites, but there are other interesting groupings we can compare, such as the genotype and the type of the sample (roots vs leaves). We could do the above all over with minor modifications, but one of the benefits of using a programming language is that you can create your own functions to automate repeated tasks. We can generalize what we did above and put it in a function like so:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">compare_alpha &lt;-<span class="st"> </span><span class="cf">function</span>(sample_data, grouping_var) {
  <span class="co"># Calcuate alpha diversity</span>
  sample_data<span class="op">$</span>alpha &lt;-<span class="st"> </span><span class="kw">diversity</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_rarefied[, sample_data<span class="op">$</span>SampleID],
                                 <span class="dt">MARGIN =</span> <span class="dv">2</span>,
                                 <span class="dt">index =</span> <span class="st">&quot;invsimpson&quot;</span>)
  
  <span class="co"># Do ANOVA</span>
  sample_data<span class="op">$</span>grouping &lt;-<span class="st"> </span>sample_data[[grouping_var]] <span class="co"># needed for how `aov` works</span>
  anova_result &lt;-<span class="st"> </span><span class="kw">aov</span>(alpha <span class="op">~</span><span class="st"> </span>grouping, sample_data)
  
  <span class="co"># Do Tukey&#39;s HSD test</span>
  tukey_result &lt;-<span class="st"> </span><span class="kw">HSD.test</span>(anova_result, <span class="st">&quot;grouping&quot;</span>, <span class="dt">group =</span> <span class="ot">TRUE</span>)

  <span class="co"># Plot result</span>
  group_data &lt;-<span class="st"> </span>tukey_result<span class="op">$</span>groups[<span class="kw">order</span>(<span class="kw">rownames</span>(tukey_result<span class="op">$</span>groups)),]
  my_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(sample_data, <span class="kw">aes</span>(<span class="dt">x =</span> grouping, <span class="dt">y =</span> alpha)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(),
              <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">rownames</span>(group_data),
                  <span class="dt">y =</span> <span class="kw">max</span>(sample_data<span class="op">$</span>alpha) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>,
                  <span class="dt">label =</span> group_data<span class="op">$</span>groups),
              <span class="dt">col =</span> <span class="st">&#39;black&#39;</span>,
              <span class="dt">size =</span> <span class="dv">10</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Alpha diversity&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(grouping_var) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Alpha diversity index&quot;</span>)
  
  <span class="co"># Return plot</span>
  <span class="kw">return</span>(my_plot)
}</code></pre></div>
<p>Using this function, we can compare plot the alpha diversities by type of sample and genotype:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">compare_alpha</span>(sample_data, <span class="st">&quot;Type&quot;</span>)</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-8-1.png" width="480" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">compare_alpha</span>(sample_data, <span class="st">&quot;Genotype&quot;</span>)</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-8-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>Looks like there is no difference in the alpha diversity between genotypes, but a large difference between the diversity of roots and leaves.</p>
<p>The <code>phyloseq</code> package (<span class="citation">McMurdie and Holmes (2013)</span>) can be used to quickly plot a variety of alpha diversity indexes per sample using the <code>plot_richness</code> function. First we need to convert the <code>taxmap</code> object to a <code>phyloseq</code> object, since all of the <code>phyloseq</code> functions expect <code>phyloseq</code> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phyloseq)</code></pre></div>
<pre><code>## 
## Attaching package: &#39;phyloseq&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:taxa&#39;:
## 
##     filter_taxa</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(metacoder)
ps_obj &lt;-<span class="st"> </span><span class="kw">as_phyloseq</span>(obj,
                      <span class="dt">otu_table =</span> <span class="st">&quot;otu_rarefied&quot;</span>,
                      <span class="dt">otu_id_col =</span> <span class="st">&quot;OTU_ID&quot;</span>,
                      <span class="dt">sample_data =</span> sample_data,
                      <span class="dt">sample_id_col =</span> <span class="st">&quot;SampleID&quot;</span>)
<span class="kw">plot_richness</span>(ps_obj, <span class="dt">color =</span> <span class="st">&quot;Type&quot;</span>, <span class="dt">x =</span> <span class="st">&quot;Site&quot;</span>)</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Each dot is a sample and the error bars in some of the indexes are the standard error.</p>
</div>
<div id="beta-between-sample-diversity" class="section level2">
<h2>Beta (between sample) diversity</h2>
<p>Beta diversity is a way to quantify the difference between two communities. There are many metrics that are used for this, but we will only mention a few of the more popular ones. A few also incorporate phylogenetic relatedness and require a phylogenetic tree of the organisms in either community to be calculated.</p>
<div id="examples-of-indexes-used-with-presenceabsence-data" class="section level4">
<h4>Examples of indexes used with presence/absence data:</h4>
<ul>
<li><strong>Sørensen</strong>: two times the number of species common to both communities divided by the sum of the number of species in each community.</li>
<li><strong>Jaccard</strong>: the number of species common to both communities divided by the number of species in either community.</li>
<li><strong>Unifrac</strong>: The fraction of the phylogenetic tree branch lengths shared by the two communities.</li>
</ul>
</div>
<div id="examples-of-indexes-used-with-count-data" class="section level4">
<h4>Examples of indexes used with count data:</h4>
<ul>
<li><strong>Bray–Curtis</strong>: The sum of lesser counts for species present in both communities divided by the sum of all counts in both communities. This can be thought of as a quantitative version of the Sørensen index.</li>
<li><strong>Weighted Unifrac</strong>: The fraction of the phylogenetic tree branch lengths shared by the two communities, weighted by the counts of organisms, so more abundant organisms have a greater influence.</li>
</ul>
<p>The <code>vegan</code> function <code>vegdist</code> is used to calculate the pairwise beta diversity indexes for a set of samples. Since this is a pairwise comparison, the output is a triangular matrix. In R, a <code>matrix</code> is like a <code>data.frame</code>, but all of the same type (e.g. all numeric), and has some different behavior.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">beta_dist &lt;-<span class="st"> </span><span class="kw">vegdist</span>(<span class="kw">t</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_rarefied[, sample_data<span class="op">$</span>SampleID]),
                     <span class="dt">index =</span> <span class="st">&quot;bray&quot;</span>)</code></pre></div>
<p>Since <code>vegdist</code> does not have a <code>MARGIN</code> option like <code>diversity</code>, we need to <a href ="00--glossary.html#transpose_anchor">transpose</a> the matrix with the <code>t</code> function.</p>
</div>
</div>
<div id="ordination" class="section level2">
<h2>Ordination</h2>
<p>Ordination is a way to display “high dimensional” data in a viable number of dimensions (2 to 3). Our data is “high dimensional” because we have many samples with many species and species can be considered a “dimension”. If we had only two species, we could make a scatter plot of their abundance in each sample and get an idea of how the samples differ. With thousands of species, this is not possible. Instead, ordination is used to try to capture the information in many dimensions by in a smaller number of new “artificial” dimensions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mds &lt;-<span class="st"> </span><span class="kw">metaMDS</span>(<span class="kw">t</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_rarefied[, sample_data<span class="op">$</span>SampleID]))</code></pre></div>
<pre class="r-output"><code>## Square root transformation
## Wisconsin double standardization
## Run 0 stress 0.08456775 
## Run 1 stress 0.09081432 
## Run 2 stress 0.1102552 
## Run 3 stress 0.1177094 
## Run 4 stress 0.1260635 
## Run 5 stress 0.0994808 
## Run 6 stress 0.1175064 
## Run 7 stress 0.1140128 
## Run 8 stress 0.1008548 
## Run 9 stress 0.100137 
## Run 10 stress 0.1103974 
## Run 11 stress 0.1157858 
## Run 12 stress 0.1079318 
## Run 13 stress 0.1139838 
## Run 14 stress 0.1081609 
## Run 15 stress 0.117441 
## Run 16 stress 0.1263571 
## Run 17 stress 0.1177241 
## Run 18 stress 0.1147656 
## Run 19 stress 0.1099166 
## Run 20 stress 0.1163623 
## *** No convergence -- monoMDS stopping criteria:
##     17: stress ratio &gt; sratmax
##      3: scale factor of the gradient &lt; sfgrmin
</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
mds_data &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(mds<span class="op">$</span>points)
mds_data<span class="op">$</span>SampleID &lt;-<span class="st"> </span><span class="kw">rownames</span>(mds_data)
mds_data &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(mds_data, sample_data)</code></pre></div>
<pre><code>## Joining, by = &quot;SampleID&quot;</code></pre>
<p>Now that we have the data in a format ggplot2 likes, we can plot it. Lets plot our two new dimensions and color them by sample type (i.e. leaves vs roots).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(mds_data, <span class="kw">aes</span>(<span class="dt">x =</span> MDS1, <span class="dt">y =</span> MDS2, <span class="dt">color =</span> Type)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>()</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This shows that leaf and root samples are quite distinct, as we would expect. We can also color them by Site:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(mds_data, <span class="kw">aes</span>(<span class="dt">x =</span> MDS1, <span class="dt">y =</span> MDS2, <span class="dt">color =</span> Site)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>()</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>It appears that within the leaf and root clusters, we “sub-clusters” corresponding to site. Finally, lets look at genotype:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(mds_data, <span class="kw">aes</span>(<span class="dt">x =</span> MDS1, <span class="dt">y =</span> MDS2, <span class="dt">color =</span> Genotype)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>()</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There is no discernible pattern there, suggesting plant genotype does not correspond to community structure.</p>
<p>We can also do the above quickly in <code>phyloseq</code> using the <code>ordinate</code> and <code>plot_ordination</code> functions. Lets look at the differences between leaf and root samples again, but using a difference index this time.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps_ord &lt;-<span class="st"> </span><span class="kw">ordinate</span>(ps_obj, <span class="dt">method =</span> <span class="st">&quot;NMDS&quot;</span>, <span class="dt">distance =</span> <span class="st">&quot;jsd&quot;</span>)</code></pre></div>
<pre class="r-output"><code>## Run 0 stress 0.1075255 
## Run 1 stress 0.107702 
## ... Procrustes: rmse 0.004332587  max resid 0.06958149 
## Run 2 stress 0.1090517 
## Run 3 stress 0.1084686 
## Run 4 stress 0.1075252 
## ... New best solution
## ... Procrustes: rmse 7.46327e-05  max resid 0.000689617 
## ... Similar to previous best
## Run 5 stress 0.1084413 
## Run 6 stress 0.1106626 
## Run 7 stress 0.1107124 
## Run 8 stress 0.1084412 
## Run 9 stress 0.1077016 
## ... Procrustes: rmse 0.004328356  max resid 0.06957641 
## Run 10 stress 0.1075833 
## ... Procrustes: rmse 0.004117664  max resid 0.06969498 
## Run 11 stress 0.1075256 
## ... Procrustes: rmse 8.493795e-05  max resid 0.0007098921 
## ... Similar to previous best
## Run 12 stress 0.1077016 
## ... Procrustes: rmse 0.004331255  max resid 0.06957523 
## Run 13 stress 0.110119 
## Run 14 stress 0.1102997 
## Run 15 stress 0.1084685 
## Run 16 stress 0.1094491 
## Run 17 stress 0.1117525 
## Run 18 stress 0.1096706 
## Run 19 stress 0.1096205 
## Run 20 stress 0.110492 
## *** Solution reached
</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_ordination</span>(ps_obj, ps_ord, <span class="dt">type =</span> <span class="st">&quot;samples&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;Type&quot;</span>)</code></pre></div>
<p><img src="07--diversity_stats_files/figure-html/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="exercises" class="section level2">
<h2>Exercises</h2>
<p><strong>1a)</strong> Look at the documentation for the <code>plot_richness</code> function from <code>phyloseq</code>. Try to make a plot using only the Simpson and inverse Simpson indexes, colored by site and split up by sample type (leaf vs root).</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-16">
Show solution
</button>
<div id="hide_buttonunnamed-chunk-16" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_richness</span>(ps_obj, <span class="dt">color =</span> <span class="st">&quot;Site&quot;</span>, <span class="dt">x =</span> <span class="st">&quot;Genotype&quot;</span>, <span class="dt">measures =</span> <span class="kw">c</span>(<span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>))</code></pre></div>
<img src="07--diversity_stats_files/figure-html/unnamed-chunk-16-1.png" width="480" style="display: block; margin: auto;" />
</div>
<p><br /></p>
<p><strong>1b)</strong> The Simpson and inverse Simpson indexes display the same information in different ways (if you know one, you can calculate the other). How do they differ?</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-17">
Show Answer
</button>
<div id="hide_buttonunnamed-chunk-17" class="collapse">
The Simpson index is bound between 0 and 1, wheras the inverse Simpson index is greater than zero. Also, the inverse Simpson index tends to spread out higher diversity values more.
</div>
<p><br /></p>
</div>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div id="ref-mcmurdie2013phyloseq">
<p>McMurdie, Paul J, and Susan Holmes. 2013. “Phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.” <em>PloS One</em> 8 (4). Public Library of Science: e61217. <a href="https://doi.org/10.1371/journal.pone.0061217" class="uri">https://doi.org/10.1371/journal.pone.0061217</a>.</p>
</div>
<div id="ref-whittaker1960vegetation">
<p>Whittaker, Robert Harding. 1960. “Vegetation of the Siskiyou Mountains, Oregon and California.” <em>Ecological Monographs</em> 30 (3). Wiley Online Library: 279–338.</p>
</div>
</div>
</div>
</div>

<div class="footer_section">
  <div class="footer_content">
    <span xmlns:dct="http://purl.org/dc/terms/" href="http://purl.org/dc/dcmitype/Text" property="dct:title" rel="dct:type">Analysis of Microbiome Community Data in R</span> by <a xmlns:cc="http://creativecommons.org/ns#" href="http://grunwaldlab.cgrb.oregonstate.edu/" property="cc:attributionName" rel="cc:attributionURL">The Grunwald lab and the USDA Horticultural Crops Research Unit</a> is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative Commons Attribution-ShareAlike 4.0 International License</a>.<br />Based on a work at <a xmlns:dct="http://purl.org/dc/terms/" href="https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r" rel="dct:source">https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r</a>.
  </div>
</div>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
