<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Introduction</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Introduction</h1>
<h3 class="subtitle"><em><em>NJ Grünwald and ZSL Foster</em></em></h3>

</div>


<pre><code>## Warning in strptime(xx, f &lt;- &quot;%Y-%m-%d %H:%M:%OS&quot;, tz = tz): unknown
## timezone &#39;zone/tz/2017c.1.0/zoneinfo/America/Los_Angeles&#39;</code></pre>
<p>This primer provides a concise introduction to conducting applied analyses of microbiome data in R based on barcodes and high throughput sequwencing (HTS). This primer does not cover metagenomic analysis that are very different in nature It is also not meant to be a textbook on statistical analysis and visualization of microbiome data. Quite to the contrary, we refer the reader to several textbooks on analysis of microbiome and ecological community data <span class="citation">(Schloss et al., 2009; Caporaso et al., 2010)</span>. Likewise, this book will not replace books on ecological theory and statistics for ecological community analysis <span class="citation">(Dixon, 2003)</span>. The reader is thus expected to have a basic understanding of ecological diversity theory and applications. The analysis presented here starts after OTU calling has occured using tools such as QIIME or mothur `r citep(bib[c(“schloss2009introducing”, “caporaso2010qiime”)]).</p>
<p>This primer is geared towards biologists interested in analyzing their microbiomes. This primer does not require extensive knowledge of programming in R, but the user is expected to install R and all packages required for this primer.</p>
<div id="why-use-r" class="section level2">
<h2>Why use R?</h2>
<p>Until recently, one of the more tedious aspects of conducting a population genetic analysis was the need for repeated reformatting data to conduct different, complimentary analyses in different programs. Often, these programs only ran on one platform. Now, R provides a toolbox with its packages that allows analysis of most data conveniently without tedious reformatting on all major computing platforms including Microsoft Windows, Linux, and Apple’s OS X. R is an open source statistical programming and graphing language that includes tools for analysis of statistical, ecological diversity and community data. This primer is for those of us that want to conduct applied analyses of microbiomes and make full use of the palette of tools available in R including for example the R markdown code for this primer.</p>
<p>Note that the R user community is very active and that both R and its packages are regularly updated, critically modified, and noted as deprecated (no longer updated) as appropriate.</p>
<blockquote>
<p>Any R user needs to make sure all components are up-to-date and that versions are compatible.</p>
</blockquote>
</div>
<div id="microbiome-data" class="section level2">
<h2>Microbiome data</h2>
<p>Before data can be anlaysed using this primer, a series of steps are typcially required:</p>
<ol style="list-style-type: decimal">
<li>Sample envoiirnments, extract DNA.</li>
<li>Barcode, pool libaries and sequence using HTS.</li>
<li>Demultiplex.</li>
<li>Call OTUs</li>
</ol>
<p>Here we focus on ths staitstical analysis and visulaizations following OTU calling:</p>
<ul>
<li>alpha and beta diversity</li>
<li>heat trees, stacked bar charts and related visualizations</li>
<li>ordination</li>
</ul>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<!--- ref --->
<div id="refs" class="references">
<div id="ref-caporaso2010qiime">
<p>Caporaso JG., Kuczynski J., Stombaugh J., Bittinger K., Bushman FD., Costello EK., Fierer N., Peña AG., Goodrich JK., Gordon JI. et al. 2010. QIIME allows analysis of high-throughput community sequencing data. <em>Nature Methods</em> 7:335–336. Available at: <a href="https://www.nature.com/articles/nmeth.f.303" class="uri">https://www.nature.com/articles/nmeth.f.303</a></p>
</div>
<div id="ref-dixon2003vegan">
<p>Dixon P. 2003. VEGAN, a package of r functions for community ecology. <em>Journal of Vegetation Science</em> 14:927–930.</p>
</div>
<div id="ref-schloss2009introducing">
<p>Schloss PD., Westcott SL., Ryabin T., Hall JR., Hartmann M., Hollister EB., Lesniewski RA., Oakley BB., Parks DH., Robinson CJ. et al. 2009. Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. <em>Applied and Environmental Microbiology</em> 75:7537–7541. Available at: <a href="http://aem.asm.org/cgi/pmidlookup?view=long&amp;pmid=19801464" class="uri">http://aem.asm.org/cgi/pmidlookup?view=long&amp;pmid=19801464</a></p>
</div>
</div>
</div>

<a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a><br /><span xmlns:dct="http://purl.org/dc/terms/" href="http://purl.org/dc/dcmitype/Text" property="dct:title" rel="dct:type">Analysis of Microbiome Community Data in R</span> by <a xmlns:cc="http://creativecommons.org/ns#" href="http://grunwaldlab.cgrb.oregonstate.edu/" property="cc:attributionName" rel="cc:attributionURL">The Grunwald lab and the USDA Horticultural Crops Research Unit</a> is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative Commons Attribution-ShareAlike 4.0 International License</a>.<br />Based on a work at <a xmlns:dct="http://purl.org/dc/terms/" href="https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r" rel="dct:source">https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r</a>.



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
