<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Data quality control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analysis of Microbiome Data in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="introduction.html">Introduction</a>
    </li>
    <li>
      <a href="quick_example.html">Example analysis</a>
    </li>
    <li>
      <a href="parsing.html">Getting data into R</a>
    </li>
    <li>
      <a href="manipulating.html">Manipulating taxonomic data</a>
    </li>
    <li>
      <a href="quality_control.html">Data quality control</a>
    </li>
    <li>
      <a href="diversity_stats.html">Diversity statistics</a>
    </li>
    <li>
      <a href="plotting.html">Plotting taxonomic data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="introduction_to_r.html">Introduction to R</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data quality control</h1>

</div>


<p>Sequencing technologies all have some amount of error. Some of this error happens during PCR and some happens during sequencing. There are lots of ways to filter out errors including:</p>
<ul>
<li>Removing sequences/bases that the sequencer indicates are low quality. This information is typically in the .fastq files returned by the sequencer. There are many programs, such as “trimmomatic”, that can use the quality scores in these files to filter out low-quality sequences.</li>
<li>Clustering similar sequences together. Many erroneous sequences are very similar to the true sequences, so they will be incorporated into the same OTU as the true sequence. In that sense, clustering into OTUs will hide some sequencing error.</li>
<li>Remove of chimeric sequences. “Chimeras” are more significant errors that occur during PCR when an incomplete amplicon acts as a primer for a different template in a subsequent cycle. Most amplicon metagenomic pipelines (e.g. QIIME, mothur, usearch) have functions to detect and remove chimeras.</li>
<li>Removing low-abundance sequences. Most of the time erroneous sequences will occur much less often than the true sequence. Simply removing any unique sequence / OTU that appears less than some minimum number of times is an effective way to remove these errors. OTUs represented by only a single sequence are commonly called “singletons”. So when you hear things like “singletons were removed” and means all OTUs with only a single sequence were removed.</li>
<li>Rarefying read counts. Usually, we try to make each sample in a sequencing run have a the same number of reads sequenced, but that does not always happen. When a sample has many more reads than another sample, its apparent diversity can be artificially inflated since rare taxa are more likely to be found. To avoid this, the reads are subsampled to a fixed number or “rarefied”.</li>
<li>Converting counts to presence/absence. Due to PCR biases and the nature of compositional data, many researchers advocate not useing read count information at all. Instead, they reccomend converting counts to simply “present” or “absent”.</li>
</ul>
<p>Since we are only working with abundance matrices in these tutorials, the first three types of quality control have already been done (ideally), but we can still remove low-abundance OTUs and rarefy the counts of the remaining OTUs.</p>
<div id="load-example-data" class="section level2">
<h2>Load example data</h2>
<p>If you are starting the workshop at this section, or had problems running code in a previous section, use the following to load the data used in this section. If <code>obj</code> and <code>sample_data</code> are already in your environment, you can ignore this and proceed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;filtered_data.Rdata&quot;</span>)</code></pre></div>
</div>
<div id="removing-low-abundance-counts" class="section level2">
<h2>Removing low-abundance counts</h2>
<p>The easiest way to get rid of some error in your data is to throw out any count information below some threshold. The threshold is up to you; removing singletons or doubletons is common, but lets be more conservative and remove any counts less than 10.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(metacoder)</code></pre></div>
<pre><code>## Loading required package: taxa</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">obj<span class="op">$</span>data<span class="op">$</span>otu_counts &lt;-<span class="st"> </span><span class="kw">zero_low_counts</span>(obj, <span class="st">&quot;otu_counts&quot;</span>, <span class="dt">min_count =</span> <span class="dv">10</span>)</code></pre></div>
<pre><code>## Converting to zero all counts less than 10.</code></pre>
<pre><code>## No `cols` specified, so using all numeric columns:
##    M1981P563, M1977P1709, M1980P502, M1981P606 ... M1955P747, M1978P342, M1955P778</code></pre>
<pre><code>## Zeroing 576512 of 9823172 counts less than 10.</code></pre>
<p>That set all read counts less than 10 to zero. It did not filter out any OTUs or their assocaited taxa however. We can do that using <code>filter_obs</code>, which is used to filter data assoicated with a taxonomy in a <code>taxmap</code> object. First lets find which OTUs now have now reads assocaited with them.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">no_reads &lt;-<span class="st"> </span><span class="kw">rowSums</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_counts[, sample_data<span class="op">$</span>SampleID]) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>
<span class="kw">sum</span>(no_reads) <span class="co"># when `sum` is used on a TRUE/FALSE vector it counts TRUEs</span></code></pre></div>
<pre><code>## [1] 26947</code></pre>
<p>So now 26947 of 33641 have no reads. We can remove them and the taxa they are assocaited with like so:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">obj &lt;-<span class="st"> </span><span class="kw">filter_obs</span>(obj, <span class="st">&quot;otu_counts&quot;</span>, <span class="op">!</span><span class="st"> </span>no_reads, <span class="dt">drop_taxa =</span> <span class="ot">TRUE</span>)
<span class="kw">print</span>(obj)</code></pre></div>
<pre><code>## &lt;Taxmap&gt;
##   489 taxa: aad. Bacteria ... axx. Fusobacteriales, bxz. Fusobacteriaceae
##   489 edges: NA-&gt;aad, aad-&gt;aaf, aaf-&gt;add ... aad-&gt;abx, abx-&gt;aja, aja-&gt;axx, axx-&gt;bxz
##   1 data sets:
##     otu_counts:
##       # A tibble: 6,694 x 293
##         taxon_id M1981P563 M1977P1709 M1980P502 M1981P606 M1981P599 M1981P611 M1981P587
##            &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##       1      ben         0          0         0         0         0         0        24
##       2      beo     10251         59     15857       309       134        90     11391
##       3      ben      1447       2496      2421      3331       143      1847      2120
##       # ... with 6,691 more rows, and 285 more variables: M1982P729 &lt;dbl&gt;, M1979P447 &lt;dbl&gt;,
##       #   M1982P701 &lt;dbl&gt;, M1982P661 &lt;dbl&gt;, M1978P293 &lt;dbl&gt;, M1980P462 &lt;dbl&gt;, M1555P221 &lt;dbl&gt;,
##       #   M1980P471 &lt;dbl&gt;, M1979P422 &lt;dbl&gt;, M1982P655 &lt;dbl&gt;, M1554P182 &lt;dbl&gt;, M1981P623 &lt;dbl&gt;,
##       #   M1555P239 &lt;dbl&gt;, M1958P1020 &lt;dbl&gt;, M1555P205 &lt;dbl&gt;, M1959P1121 &lt;dbl&gt;, M1551P83 &lt;dbl&gt;,
##       #   M1961P1359 &lt;dbl&gt;, M1551P21 &lt;dbl&gt;, M1690P1399 &lt;dbl&gt;, M1555P214 &lt;dbl&gt;, M1955P819 &lt;dbl&gt;,
##       #   M1979P426 &lt;dbl&gt;, M1957P982 &lt;dbl&gt;, M1979P369 &lt;dbl&gt;, M1690P1397 &lt;dbl&gt;, M1978P318 &lt;dbl&gt;,
##       #   M1956P853 &lt;dbl&gt;, M1979P381 &lt;dbl&gt;, M1960P1219 &lt;dbl&gt;, M1554P112 &lt;dbl&gt;, M1980P533 &lt;dbl&gt;,
##       #   M1554P138 &lt;dbl&gt;, M1982P711 &lt;dbl&gt;, M1978P307 &lt;dbl&gt;, M1982P696 &lt;dbl&gt;, M1555P231 &lt;dbl&gt;,
##       #   M1980P520 &lt;dbl&gt;, M1555P190 &lt;dbl&gt;, M1955P739 &lt;dbl&gt;, M1981P578 &lt;dbl&gt;, M1980P506 &lt;dbl&gt;,
##       #   M1981P631 &lt;dbl&gt;, M1980P461 &lt;dbl&gt;, M1982P667 &lt;dbl&gt;, M1980P496 &lt;dbl&gt;, M1982P707 &lt;dbl&gt;,
##       #   M1980P469 &lt;dbl&gt;, M1980P486 &lt;dbl&gt;, M1980P532 &lt;dbl&gt;, M1977P1714 &lt;dbl&gt;, M1981P597 &lt;dbl&gt;,
##       #   M1977P1705 &lt;dbl&gt;, M1955P764 &lt;dbl&gt;, M1981P588 &lt;dbl&gt;, M1982P709 &lt;dbl&gt;, M1980P475 &lt;dbl&gt;,
##       #   M1980P539 &lt;dbl&gt;, M1982P645 &lt;dbl&gt;, M1981P577 &lt;dbl&gt;, M1980P535 &lt;dbl&gt;, M1981P609 &lt;dbl&gt;,
##       #   M1554P132 &lt;dbl&gt;, M1981P556 &lt;dbl&gt;, M1551P92 &lt;dbl&gt;, M1981P638 &lt;dbl&gt;, M1551P67 &lt;dbl&gt;,
##       #   M1979P439 &lt;dbl&gt;, M1551P37 &lt;dbl&gt;, M1981P586 &lt;dbl&gt;, M1555P256 &lt;dbl&gt;, M1961P1298 &lt;dbl&gt;,
##       #   M1554P124 &lt;dbl&gt;, M1955P814 &lt;dbl&gt;, M1555P238 &lt;dbl&gt;, M1961P1308 &lt;dbl&gt;, M1979P378 &lt;dbl&gt;,
##       #   M1982P705 &lt;dbl&gt;, M1555P212 &lt;dbl&gt;, M1980P481 &lt;dbl&gt;, M1979P383 &lt;dbl&gt;, M1981P632 &lt;dbl&gt;,
##       #   M1555P240 &lt;dbl&gt;, M1981P561 &lt;dbl&gt;, M1554P128 &lt;dbl&gt;, M1982P664 &lt;dbl&gt;, M1978P356 &lt;dbl&gt;,
##       #   M1980P538 &lt;dbl&gt;, M1980P499 &lt;dbl&gt;, M1982P683 &lt;dbl&gt;, M1980P507 &lt;dbl&gt;, M1980P501 &lt;dbl&gt;,
##       #   M1955P745 &lt;dbl&gt;, M1977P1711 &lt;dbl&gt;, M1980P476 &lt;dbl&gt;, M1982P646 &lt;dbl&gt;, M1982P703 &lt;dbl&gt;,
##       #   M1981P584 &lt;dbl&gt;, M1980P470 &lt;dbl&gt;, M1956P899 &lt;dbl&gt;, ...
##   0 functions:</code></pre>
</div>
<div id="rarefaction" class="section level2">
<h2>Rarefaction</h2>
<p>Rarefaction is used to simulate even sampling depth. Even sampling depth is important for at least two reasons:</p>
<ul>
<li>When comparing diversity of samples, a higher sampling depth can inflate appparent diverstiy since more rare species are likely to be found. The effect of this will depend on the diversity index used.</li>
<li>When comparing the similarity of samples, the presence of rare species due to higher sampling depth in one sample but not another can make the two samples appear more differnet than they actually are.</li>
</ul>
<p>Therefore, when comparing the diversity or similarity of samples, it is important to rarefy, or subsample, to a constant depth. Typically, the depth chosen is the minimum sample depth. If the minimum depth is very small, the samples with the smallest depth can be removed and the minimum depth of the remaining samples can be used.</p>
<p>Lets take a look at the distribution of read depths of our samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(<span class="kw">colSums</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_counts[, <span class="op">-</span><span class="dv">1</span>]))</code></pre></div>
<p><img src="quality_control_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>We have a minimum depth of 1857, a median of 78767 and a maximum depth of 1.8935310^{5}. We could try to remove one or two of the samples with the smallest depth, since it seems like a waste to throw out so much data, but for this tutorial we will just rarefy to the minimum of 1857 reads. The <code>vegan</code> package implements many functions to help with rarefying data, but we will use a function from <code>metacoder</code> that calls the functions from <code>vegan</code> and reformats the input and ouputs to make them easier to use with the way our data is formatted.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">obj<span class="op">$</span>data<span class="op">$</span>otu_rarefied &lt;-<span class="st"> </span><span class="kw">rarefy_obs</span>(obj, <span class="st">&quot;otu_counts&quot;</span>)</code></pre></div>
<pre><code>## No `cols` specified, so using all numeric columns:
##    M1981P563, M1977P1709, M1980P502, M1981P606 ... M1955P747, M1978P342, M1955P778</code></pre>
<pre><code>## Rarefying to 1857 since that is the lowest sample total.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(obj)</code></pre></div>
<pre><code>## &lt;Taxmap&gt;
##   489 taxa: aad. Bacteria ... axx. Fusobacteriales, bxz. Fusobacteriaceae
##   489 edges: NA-&gt;aad, aad-&gt;aaf, aaf-&gt;add ... aad-&gt;abx, abx-&gt;aja, aja-&gt;axx, axx-&gt;bxz
##   2 data sets:
##     otu_counts:
##       # A tibble: 6,694 x 293
##         taxon_id M1981P563 M1977P1709 M1980P502 M1981P606 M1981P599 M1981P611 M1981P587
##            &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##       1      ben         0          0         0         0         0         0        24
##       2      beo     10251         59     15857       309       134        90     11391
##       3      ben      1447       2496      2421      3331       143      1847      2120
##       # ... with 6,691 more rows, and 285 more variables: M1982P729 &lt;dbl&gt;, M1979P447 &lt;dbl&gt;,
##       #   M1982P701 &lt;dbl&gt;, M1982P661 &lt;dbl&gt;, M1978P293 &lt;dbl&gt;, M1980P462 &lt;dbl&gt;, M1555P221 &lt;dbl&gt;,
##       #   M1980P471 &lt;dbl&gt;, M1979P422 &lt;dbl&gt;, M1982P655 &lt;dbl&gt;, M1554P182 &lt;dbl&gt;, M1981P623 &lt;dbl&gt;,
##       #   M1555P239 &lt;dbl&gt;, M1958P1020 &lt;dbl&gt;, M1555P205 &lt;dbl&gt;, M1959P1121 &lt;dbl&gt;, M1551P83 &lt;dbl&gt;,
##       #   M1961P1359 &lt;dbl&gt;, M1551P21 &lt;dbl&gt;, M1690P1399 &lt;dbl&gt;, M1555P214 &lt;dbl&gt;, M1955P819 &lt;dbl&gt;,
##       #   M1979P426 &lt;dbl&gt;, M1957P982 &lt;dbl&gt;, M1979P369 &lt;dbl&gt;, M1690P1397 &lt;dbl&gt;, M1978P318 &lt;dbl&gt;,
##       #   M1956P853 &lt;dbl&gt;, M1979P381 &lt;dbl&gt;, M1960P1219 &lt;dbl&gt;, M1554P112 &lt;dbl&gt;, M1980P533 &lt;dbl&gt;,
##       #   M1554P138 &lt;dbl&gt;, M1982P711 &lt;dbl&gt;, M1978P307 &lt;dbl&gt;, M1982P696 &lt;dbl&gt;, M1555P231 &lt;dbl&gt;,
##       #   M1980P520 &lt;dbl&gt;, M1555P190 &lt;dbl&gt;, M1955P739 &lt;dbl&gt;, M1981P578 &lt;dbl&gt;, M1980P506 &lt;dbl&gt;,
##       #   M1981P631 &lt;dbl&gt;, M1980P461 &lt;dbl&gt;, M1982P667 &lt;dbl&gt;, M1980P496 &lt;dbl&gt;, M1982P707 &lt;dbl&gt;,
##       #   M1980P469 &lt;dbl&gt;, M1980P486 &lt;dbl&gt;, M1980P532 &lt;dbl&gt;, M1977P1714 &lt;dbl&gt;, M1981P597 &lt;dbl&gt;,
##       #   M1977P1705 &lt;dbl&gt;, M1955P764 &lt;dbl&gt;, M1981P588 &lt;dbl&gt;, M1982P709 &lt;dbl&gt;, M1980P475 &lt;dbl&gt;,
##       #   M1980P539 &lt;dbl&gt;, M1982P645 &lt;dbl&gt;, M1981P577 &lt;dbl&gt;, M1980P535 &lt;dbl&gt;, M1981P609 &lt;dbl&gt;,
##       #   M1554P132 &lt;dbl&gt;, M1981P556 &lt;dbl&gt;, M1551P92 &lt;dbl&gt;, M1981P638 &lt;dbl&gt;, M1551P67 &lt;dbl&gt;,
##       #   M1979P439 &lt;dbl&gt;, M1551P37 &lt;dbl&gt;, M1981P586 &lt;dbl&gt;, M1555P256 &lt;dbl&gt;, M1961P1298 &lt;dbl&gt;,
##       #   M1554P124 &lt;dbl&gt;, M1955P814 &lt;dbl&gt;, M1555P238 &lt;dbl&gt;, M1961P1308 &lt;dbl&gt;, M1979P378 &lt;dbl&gt;,
##       #   M1982P705 &lt;dbl&gt;, M1555P212 &lt;dbl&gt;, M1980P481 &lt;dbl&gt;, M1979P383 &lt;dbl&gt;, M1981P632 &lt;dbl&gt;,
##       #   M1555P240 &lt;dbl&gt;, M1981P561 &lt;dbl&gt;, M1554P128 &lt;dbl&gt;, M1982P664 &lt;dbl&gt;, M1978P356 &lt;dbl&gt;,
##       #   M1980P538 &lt;dbl&gt;, M1980P499 &lt;dbl&gt;, M1982P683 &lt;dbl&gt;, M1980P507 &lt;dbl&gt;, M1980P501 &lt;dbl&gt;,
##       #   M1955P745 &lt;dbl&gt;, M1977P1711 &lt;dbl&gt;, M1980P476 &lt;dbl&gt;, M1982P646 &lt;dbl&gt;, M1982P703 &lt;dbl&gt;,
##       #   M1981P584 &lt;dbl&gt;, M1980P470 &lt;dbl&gt;, M1956P899 &lt;dbl&gt;, ...
##     otu_rarefied:
##       # A tibble: 6,694 x 293
##         taxon_id M1981P563 M1977P1709 M1980P502 M1981P606 M1981P599 M1981P611 M1981P587
##       *    &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##       1      ben         0          0         0         0         0         0         0
##       2      beo       362          0       414         9       134         2       405
##       3      ben        44         31        79        61       143        40        95
##       # ... with 6,691 more rows, and 285 more variables: M1982P729 &lt;dbl&gt;, M1979P447 &lt;dbl&gt;,
##       #   M1982P701 &lt;dbl&gt;, M1982P661 &lt;dbl&gt;, M1978P293 &lt;dbl&gt;, M1980P462 &lt;dbl&gt;, M1555P221 &lt;dbl&gt;,
##       #   M1980P471 &lt;dbl&gt;, M1979P422 &lt;dbl&gt;, M1982P655 &lt;dbl&gt;, M1554P182 &lt;dbl&gt;, M1981P623 &lt;dbl&gt;,
##       #   M1555P239 &lt;dbl&gt;, M1958P1020 &lt;dbl&gt;, M1555P205 &lt;dbl&gt;, M1959P1121 &lt;dbl&gt;, M1551P83 &lt;dbl&gt;,
##       #   M1961P1359 &lt;dbl&gt;, M1551P21 &lt;dbl&gt;, M1690P1399 &lt;dbl&gt;, M1555P214 &lt;dbl&gt;, M1955P819 &lt;dbl&gt;,
##       #   M1979P426 &lt;dbl&gt;, M1957P982 &lt;dbl&gt;, M1979P369 &lt;dbl&gt;, M1690P1397 &lt;dbl&gt;, M1978P318 &lt;dbl&gt;,
##       #   M1956P853 &lt;dbl&gt;, M1979P381 &lt;dbl&gt;, M1960P1219 &lt;dbl&gt;, M1554P112 &lt;dbl&gt;, M1980P533 &lt;dbl&gt;,
##       #   M1554P138 &lt;dbl&gt;, M1982P711 &lt;dbl&gt;, M1978P307 &lt;dbl&gt;, M1982P696 &lt;dbl&gt;, M1555P231 &lt;dbl&gt;,
##       #   M1980P520 &lt;dbl&gt;, M1555P190 &lt;dbl&gt;, M1955P739 &lt;dbl&gt;, M1981P578 &lt;dbl&gt;, M1980P506 &lt;dbl&gt;,
##       #   M1981P631 &lt;dbl&gt;, M1980P461 &lt;dbl&gt;, M1982P667 &lt;dbl&gt;, M1980P496 &lt;dbl&gt;, M1982P707 &lt;dbl&gt;,
##       #   M1980P469 &lt;dbl&gt;, M1980P486 &lt;dbl&gt;, M1980P532 &lt;dbl&gt;, M1977P1714 &lt;dbl&gt;, M1981P597 &lt;dbl&gt;,
##       #   M1977P1705 &lt;dbl&gt;, M1955P764 &lt;dbl&gt;, M1981P588 &lt;dbl&gt;, M1982P709 &lt;dbl&gt;, M1980P475 &lt;dbl&gt;,
##       #   M1980P539 &lt;dbl&gt;, M1982P645 &lt;dbl&gt;, M1981P577 &lt;dbl&gt;, M1980P535 &lt;dbl&gt;, M1981P609 &lt;dbl&gt;,
##       #   M1554P132 &lt;dbl&gt;, M1981P556 &lt;dbl&gt;, M1551P92 &lt;dbl&gt;, M1981P638 &lt;dbl&gt;, M1551P67 &lt;dbl&gt;,
##       #   M1979P439 &lt;dbl&gt;, M1551P37 &lt;dbl&gt;, M1981P586 &lt;dbl&gt;, M1555P256 &lt;dbl&gt;, M1961P1298 &lt;dbl&gt;,
##       #   M1554P124 &lt;dbl&gt;, M1955P814 &lt;dbl&gt;, M1555P238 &lt;dbl&gt;, M1961P1308 &lt;dbl&gt;, M1979P378 &lt;dbl&gt;,
##       #   M1982P705 &lt;dbl&gt;, M1555P212 &lt;dbl&gt;, M1980P481 &lt;dbl&gt;, M1979P383 &lt;dbl&gt;, M1981P632 &lt;dbl&gt;,
##       #   M1555P240 &lt;dbl&gt;, M1981P561 &lt;dbl&gt;, M1554P128 &lt;dbl&gt;, M1982P664 &lt;dbl&gt;, M1978P356 &lt;dbl&gt;,
##       #   M1980P538 &lt;dbl&gt;, M1980P499 &lt;dbl&gt;, M1982P683 &lt;dbl&gt;, M1980P507 &lt;dbl&gt;, M1980P501 &lt;dbl&gt;,
##       #   M1955P745 &lt;dbl&gt;, M1977P1711 &lt;dbl&gt;, M1980P476 &lt;dbl&gt;, M1982P646 &lt;dbl&gt;, M1982P703 &lt;dbl&gt;,
##       #   M1981P584 &lt;dbl&gt;, M1980P470 &lt;dbl&gt;, M1956P899 &lt;dbl&gt;, ...
##   0 functions:</code></pre>
</div>
<div id="rarefaction-curves" class="section level2">
<h2>Rarefaction curves</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vegan)</code></pre></div>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.4-3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rarecurve</span>(<span class="kw">t</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_counts[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="op">-</span><span class="dv">1</span>]), <span class="dt">step =</span> <span class="dv">20</span>,
          <span class="dt">sample =</span> <span class="kw">min</span>(<span class="kw">colSums</span>(obj<span class="op">$</span>data<span class="op">$</span>otu_counts[, <span class="op">-</span><span class="dv">1</span>])), <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="dt">cex =</span> <span class="fl">0.6</span>)</code></pre></div>
<p><img src="quality_control_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="converting-to-presenceabsence" class="section level2">
<h2>Converting to presence/absence</h2>
</div>
<div id="saving-the-data-for-the-next-section" class="section level2">
<h2>Saving the data for the next section</h2>
<p>In the next section we will work on analyzing this data set. Don’t worry about running the following code; it is just there to let us use the same data set on multiple website pages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">save</span>(obj, sample_data, <span class="dt">file =</span> <span class="st">&quot;clean_data.Rdata&quot;</span>)</code></pre></div>
</div>

<a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a><br /><span xmlns:dct="http://purl.org/dc/terms/" href="http://purl.org/dc/dcmitype/Text" property="dct:title" rel="dct:type">Analysis of Microbiome Community Data in R</span> by <a xmlns:cc="http://creativecommons.org/ns#" href="http://grunwaldlab.cgrb.oregonstate.edu/" property="cc:attributionName" rel="cc:attributionURL">The Grunwald lab and the USDA Horticultural Crops Research Unit</a> is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative Commons Attribution-ShareAlike 4.0 International License</a>.<br />Based on a work at <a xmlns:dct="http://purl.org/dc/terms/" href="https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r" rel="dct:source">https://github.com/grunwaldlab/analysis_of_microbiome_community_data_in_r</a>.



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
